Assets/Scripts/Doll.cs

```diff
@@ -5,9 +5,28 @@ public class Doll : MonoBehaviour
     [SerializeField] private Renderer _headRenderer;
     [SerializeField] private Renderer _bodyRenderer;
 
+    private float _minZ;
+    private float _maxZ;
+
     public void SetColor(Color color)
     {
         _headRenderer.material.color = color;
         _bodyRenderer.material.color = color;
     }
+
+    public void SetMoveRange(float minZ, float maxZ)
+    {
+        _minZ = minZ;
+        _maxZ = maxZ;
+    }
+
+    public (float minZ, float maxZ) GetMoveRange()
+    {
+        return (_minZ, _maxZ);
+    }

```

Comment:
範囲を公開するのではなく、機能を提供してあげたほうがよさそう。
```suggestion
    public bool IsWithinMoveRange(float z)
    {
        return moveRange.minZ < z && z < moveRange.maxZ
    }
```